frontend/javascripts/oxalis/shaders/texture_access.glsl.ts

```diff
@@ -86,16 +88,85 @@ export const getColorForCoords: ShaderModule = {
     linearizeVec3ToIndexWithMod,
     getRgbaAtIndex,
     getRgbaAtXYIndex,
-    getRelativeCoords,
+    getAbsoluteCoords,
     getResolutionFactors,
+    hashCombine,
   ],
   code: `
+    float NOT_YET_COMMITTED_VALUE = pow(2., 21.) - 1.;
+
+    float attemptLookUpLookUp(uint globalLayerIndex, uvec4 bucketAddress, uint seed) {
+      <% if (!isFragment) { %>
+        outputSeed[globalLayerIndex] = seed;
+      <% } %>
+
+
+      highp uint h0 = hashCombine(seed, bucketAddress.x);
+      h0 = hashCombine(h0, bucketAddress.y);
+      h0 = hashCombine(h0, bucketAddress.z);
+      h0 = hashCombine(h0, bucketAddress.a);
+      h0 = hashCombine(h0, globalLayerIndex);
+      h0 = h0 % LOOKUP_CUCKOO_ENTRY_CAPACITY;
+      h0 = uint(h0 * LOOKUP_CUCKOO_ELEMENTS_PER_ENTRY / LOOKUP_CUCKOO_ELEMENTS_PER_TEXEL);
+
+      highp uint x = h0 % LOOKUP_CUCKOO_TWIDTH;
+      highp uint y = h0 / LOOKUP_CUCKOO_TWIDTH;
+
+      uvec4 compressedEntry = texelFetch(lookup_texture, ivec2(x, y), 0);
+
+      <% if (!isFragment) { %>
+        outputCompressedEntry[globalLayerIndex] = compressedEntry;
+      <% } %>
+
+      uint compressedBytes = compressedEntry.a;
+      uint foundMagIdx = compressedBytes >> (32u - 5u);
+      uint foundLayerIndex = (compressedBytes >> 21u) & (uint(pow(2., 6.)) - 1u);
+
+      if (compressedEntry.xyz != bucketAddress.xyz
+        || globalLayerIndex != foundLayerIndex
+        || foundMagIdx != bucketAddress.a) {
+        <% if (!isFragment) { %>
+          outputAddress[globalLayerIndex] = -1.;
+        <% } %>
+        return -1.;
+      }
+      uint address = compressedBytes & (uint(pow(2., 21.)) - 1u);
+
+      <% if (!isFragment) { %>
+        outputAddress[globalLayerIndex] = float(address);
+      <% } %>
+
+      return float(address);
+    }
+
+    float lookUpBucket(uint globalLayerIndex, uvec4 bucketAddress, bool supportsPrecomputedBucketAddress) {
+      // The fragment shader can read the entry that was
+      // calculated by the vertex shader. However, this won't always
+      // be precise because the triangles of the plane won't necessarily
+      // align with the bucket borders.
+      <% if (isFragment) { %>
+        if (supportsPrecomputedBucketAddress) {
```

Comment:
I did not find an explanation for this parameter. I expect it to be related to the triangle border artifacts you experienced in an intermediate version. Is that correct?

Reply:
`supportsPrecomputedBucketAddress` is true if a layer doesn't have special transforms and if interpolation is not used. Only in these cases, the bucket address that was precomputed in the vertex shader can be used.